core/src/scene/sceneLoader.cpp

```diff
@@ -990,6 +992,24 @@ void SceneLoader::loadSource(const std::shared_ptr<Platform>& platform, const st
         url = urlStream.str();
     }
 
+    auto constructUrls = [&](int loc, const std::string& subD) {
+        urls.push_back(url);
+        urls.back().replace(loc, 3, subD);
+    };
+
+    if (Node subDomainNode = source["url_subdomains"]) {
+        auto loc = url.find("{s}");
+        if (loc != std::string::npos && subDomainNode.IsSequence()) {
+            for (const auto& domain : subDomainNode) {
+                if (domain.IsScalar()) {
+                    constructUrls(loc, domain.Scalar());
```

Comment:
Its not doing too much work irrespective. It feels you are strong on this so will change it. Stay tuned.